chore(android): Add missing gradle configuration options (#8301)
getsentry/sentry-docs: src/platforms/android/configuration/gradle.mdx

## ProGuard/R8 & DexGuard

> The plugin will automatically generate appropriate ProGuard mapping files and upload them when you run `gradle assemble{BuildVariant}`. For example, `assembleRelease` — Release is the default, but the plugin works for others if you have enabled ProGuard/R8. The credentials for the upload step are defined in the Gradle extension above or loaded from a `sentry.properties` file in your project.

Reviewer: ```suggestion

The plugin will automatically generate appropriate ProGuard mapping files and upload them when you run `gradle assemble{BuildVariant}`. For example, `assembleRelease`. Release is the default, but the plugin works for others as long as you've enabled ProGuard/R8. The credentials for the upload step are defined in the Gradle extension above or loaded from a `sentry.properties` file in your project. 

```

> If you are using the Guardsquare's ProGuard or DexGuard tooling, you have to enable the `experimentalGuardsquareSupport` flag.
> 
> Learn more in the [full sentry-cli documentation](/product/cli/configuration/#configuration-values).

Reviewer: ```suggestion
To use Guardsquare's ProGuard or DexGuard tooling, you'll have to enable the `experimentalGuardsquareSupport` flag. Learn more in the [full sentry-cli documentation](/product/cli/configuration/#configuration-values).
```

> Alternatively to providing credentials via Gradle extension, you can create a `sentry.properties` file with the following contents under your `app` module:

Reviewer: ```suggestion
Instead of providing your credentials via the Gradle extension, you can create a `sentry.properties` file with the following content under your `app` module:
```
